lib/stormpath-sdk/client.rb

```diff
@@ -20,18 +20,18 @@ class Client
     include Stormpath::Util::Assert
     include Stormpath::Resource::Associations
 
-    attr_reader :data_store, :application
+    attr_reader :data_store, :application, :api_key
```

Comment:
Oh yes, I see that you wouldn't be able to see the api key from the application class like this.

However, I still think we shouldn't have it in the client. The way we've done it in other SDKs is that we add it to the data store and call it from there. This approach is perfect in languages where you can have internal interfaces, but I think it still applies for Ruby as we don't expose the api key directly on the client. This will require that we don't store the api key in the `HttpClientRequestExecutor` class anymore and just pass it as a parameter on every request (as part of the request object) from the data store.

Let me know if you have any questions.